src/OrchardCore.Modules/OrchardCore.Title/Drivers/TitlePartDisplayDriver.cs

```diff
@@ -39,7 +39,7 @@ public override IDisplayResult Edit(TitlePart titlePart, BuildPartEditorContext
     {
         return Initialize<TitlePartViewModel>(GetEditorShapeType(context), model =>
         {
-            model.Title = titlePart.ContentItem.DisplayText;
+            model.Title = titlePart.Title;
```

Comment:
Makes sense for me, why the `TitlePart` needs to rely on the `DisplayText`, which could be changed from the `ContentHandler` or programatically, which leads to inconsistency

/cc @Piedone